evosax/subpops/protocol.py

```diff
@@ -31,3 +35,22 @@ def independent(
     ) -> Tuple[chex.Array, chex.ArrayTree]:
         """Simply return non-altered candidates & fitness."""
         return batch_x, batch_fitness
+
+
+    @partial(jax.jit, static_argnums=(0,))
+    def best_subpop(
+        self, batch_x: chex.Array, batch_fitness: chex.Array
+    ) -> Tuple[chex.Array, chex.ArrayTree]:
+        """Find the subpop with the globally best candidate and set all subpops
+        to the same as """
+        global_best_arg = batch_fitness.argmin()
+        best_subpop_ind = jnp.unravel_index(global_best_arg, batch_fitness.shape)[0]
+
+        best_subpop_x =  batch_x[best_subpop_ind]
+        best_subpop_fitness =  batch_fitness[best_subpop_ind]
+
+        for i in range(batch_x.shape[0]):
```

Comment:
I don't think we want to replace all of the gathered fitness information by the best overall score/candidate, since this will discard a lot of information about the search space that is potentially useful. Instead, I would propose to only replace the worst performer in each subpopulation with the globally best one.

Reply:
Alternatively, we could have a hyperparameter `top_k`, which performs this replace for the best/worst k population members.

Reply:
I agree that a "global best" is more interesting, but I had a couple concerns with how to implement this in a jittable way. I think it might be easiest to chat about via another format, but if you want to take a look at some initial code in that direction, here is a PR:
https://github.com/DiamonDiva/evosax/pull/2/files

The protocol implemented here is much simpler and is easily jittable (as far as I can tell), and so that's why it's the only one with which I've made a direct PR so far.

Lastly, the `top_k` concept is ultimately the long-term flexible solution in my mind too, and should be a simple follow-up after sorting out how to efficiently share the best member, caring to not have duplicates